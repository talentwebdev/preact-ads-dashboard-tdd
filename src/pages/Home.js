import React from 'react'
import {Grid} from '@material-ui/core';
import AdminDashboard from '../sidepages/AdminDashboard';
import '../styles/Home.css';
import {Link} from 'react-router-dom';
function Home() {
    return (
        <div className="home">
            <div className="sideBar">
                <div className="sideBarTitle">
                    <p>BHHS Broker Tools</p>
                </div>
                <div className="sideBarMenu">
                    <Link to="/">Home</Link>
                    <Link to="/">Admin Dashboard</Link>
                    <Link to="/">BOOST Statistics</Link>
                    <Link to="/">InstaView Statistics</Link>
                    <Link to="/">Agent Profile</Link>
                    <Link to="/">Agent Search</Link>
                    <Link to="/">Listing Search</Link>
                    <Link to="/">All Tours</Link>
                    <Link to="/">Training</Link>
                    <Link to="/">Marketing Materials</Link>
                    <Link to="/">Help/Contact Us</Link>
                </div>
            </div>
            <div className="mainBody">
                <AdminDashboard />
            </div>
            {/* <Grid item lg={12} container spacing={2}>
                <Grid item lg={3} md={4} sm={5} xs={12}>
                    
                </Grid>
                <Grid item lg={9} md={8} sm={7} xs={12}>
                    
                </Grid>
            </Grid> */}
        </div>
    )
}

export default Home;
