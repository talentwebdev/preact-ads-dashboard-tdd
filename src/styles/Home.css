.itemContent,.itemFooter,.itemFooter .itemFooterTitle,.itemFooter .itemFooterIcon{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.sideBar,.sideBarTitle,.sideBarMenu,.adminMainBody .viewItem .itemTitle{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.itemContent .itemContentLeft p,.itemContent .itemContentRight p,.itemContent .itemContentLeft h3,.itemContent .itemContentRight h3{text-align:left;margin-left:5px;line-height:5px}.itemContent .itemContentLeft,.itemContent .itemContentRight{height:4.5vw;max-height:20vw;margin-top:1vw;margin-left:5px;margin-right:5px;width:50%;padding:5px 0;border-radius:10px;-webkit-box-shadow:0px 3px 15px rgba(0,0,0,0.5);box-shadow:0px 3px 15px rgba(0,0,0,0.5)}.itemContent .itemContentLeft p,.itemContent .itemContentRight p{font-size:0.8vw}.itemContent .itemContentLeft h3,.itemContent .itemContentRight h3{font-size:1.2vw}*{-webkit-box-sizing:border-box;box-sizing:border-box}.home{display:-webkit-box;display:-ms-flexbox;display:flex;width:72%;margin:auto;margin-top:2vw}.sideBar{background-color:#fff;-webkit-box-flex:22%;-ms-flex:22%;flex:22%;border:2px solid #0f4791;border-radius:5px;max-height:80vw;height:25vw;margin-top:4px}.mainBody{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-flex:78%;-ms-flex:78%;flex:78%;-ms-flex-wrap:wrap;flex-wrap:wrap}.sideBarTitle{margin-top:0;background-color:#4d1936;height:2.5vw;max-height:10vw;width:100%;overflow:hidden;border-top-left-radius:3px;border-top-right-radius:3px}.sideBarTitle p{color:#fff;font-weight:600;font-size:1.2vw}.sideBarMenu{height:100%;width:100%}.sideBarMenu a{color:#4D1936;text-decoration:none;font-size:1.3vw;font-family:'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;padding:0.2vw}.sideBarMenu a:hover{background-color:gray;width:100%}.adminMainBody{display:-ms-grid;display:grid;-ms-grid-columns:1fr 1fr 1fr;grid-template-columns:1fr 1fr 1fr;place-items:center;height:auto;-webkit-box-align:center;-ms-flex-align:center;align-items:center;place-items:center}.adminMainBody .items{width:18vw;height:12.4vw;border:2px solid #0f4791;border-radius:10px;margin:3px;-webkit-box-shadow:0px 3px 15px rgba(0,0,0,0.5);box-shadow:0px 3px 15px rgba(0,0,0,0.5)}.adminMainBody .viewItem{width:18vw;height:12.4vw;border:2px solid #0f4791;border-radius:10px;margin:3px;-webkit-box-shadow:0px 3px 15px rgba(0,0,0,0.5);box-shadow:0px 3px 15px rgba(0,0,0,0.5);overflow:hidden}.adminMainBody .viewItem .itemTitle{background-color:#4D1936;height:2.4vw}.adminMainBody .viewItem .itemTitle p{color:#fff;font-size:1.2vw;font-weight:600}.itemFooter{height:2.5vw;background-color:#E5D9B3;margin-top:1vw}.itemFooter .itemFooterTitle{color:#4D1936;-webkit-box-pack:end;-ms-flex-pack:end;justify-content:flex-end;-webkit-box-flex:70%;-ms-flex:70%;flex:70%;font-size:1.4vw;font-weight:600;font-family:'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;padding:0.2vw}.itemFooter .itemFooterIcon{-webkit-box-flex:30%;-ms-flex:30%;flex:30%;color:#4D1936}@media only screen and (max-width: 1450px){.sideBar{height:35vw}.sideBarMenu{width:19vw}.sideBarTitle{height:3vw}.sideBarTitle p{font-size:1.7vw}.sideBar .sideBarMenu a{font-size:1.5vw;padding:0.5vw}.adminMainBody{-ms-grid-columns:1fr 1fr;grid-template-columns:1fr 1fr}.adminMainBody .items{width:26vw;height:18vw}.adminMainBody .viewItem{width:26vw;height:18vw}.adminMainBody .viewItem .itemTitle{height:3vw}.adminMainBody .viewItem .itemTitle p{font-size:1.7vw}.adminMainBody .viewItem .itemContentLeft{height:7vw}.adminMainBody .viewItem .itemContentLeft p{font-size:1.2vw}.adminMainBody .viewItem .itemContentLeft h3{font-size:1.8vw}.adminMainBody .viewItem .itemContentRight{height:7vw}.adminMainBody .viewItem .itemContentRight p{font-size:1.2vw}.adminMainBody .viewItem .itemContentRight h3{font-size:1.8vw}.adminMainBody .viewItem .itemFooter{height:4vw}.adminMainBody .viewItem .itemFooter .itemFooterTitle{font-size:2vw}}@media only screen and (max-width: 1250px){.sideBar .sideBarMenu a{font-size:1.5vw}.adminMainBody{-ms-grid-columns:1fr 1fr;grid-template-columns:1fr 1fr}.adminMainBody .items{width:26vw;height:15vw}.adminMainBody .viewItem{width:26vw;height:15vw}.adminMainBody .viewItem .itemContentLeft{height:5vw}.adminMainBody .viewItem .itemContentLeft p{font-size:1.0vw}.adminMainBody .viewItem .itemContentLeft h3{font-size:1.5vw}.adminMainBody .viewItem .itemContentRight{height:5vw}.adminMainBody .viewItem .itemContentRight p{font-size:1.0vw}.adminMainBody .viewItem .itemContentRight h3{font-size:1.5vw}.adminMainBody .viewItem .itemFooter{height:3.5vw}.adminMainBody .viewItem .itemFooter .itemFooterTitle{font-size:2vw}}@media only screen and (max-width: 950px){.sideBar{height:50vw;margin-left:2vw}.sideBarTitle{height:5vw}.sideBarTitle p{font-size:2.5vw}.sideBarMenu{width:27vw}.sideBar .sideBarMenu a{font-size:2.5vw}.adminMainBody{-ms-grid-columns:1fr;grid-template-columns:1fr}.adminMainBody .items{width:37vw;height:27vw}.adminMainBody .viewItem{width:37vw;height:27vw}.adminMainBody .viewItem .itemTitle{height:5vw}.adminMainBody .viewItem .itemTitle p{font-size:2.5vw}.adminMainBody .viewItem .itemContentLeft{height:10vw;margin-top:1.2vw 0vw;margin-bottom:0.5vw}.adminMainBody .viewItem .itemContentLeft p{font-size:1.8vw}.adminMainBody .viewItem .itemContentLeft h3{font-size:2.5vw}.adminMainBody .viewItem .itemContentRight{height:10vw;margin-top:1.2vw;margin-bottom:0.5vw}.adminMainBody .viewItem .itemContentRight p{font-size:1.8vw}.adminMainBody .viewItem .itemContentRight h3{font-size:2.5vw}.adminMainBody .viewItem .itemFooter{height:6vw}.adminMainBody .viewItem .itemFooter .itemFooterTitle{font-size:3vw}}@media only screen and (max-width: 600px){.home{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column}.home .sideBar{-webkit-box-flex:100%;-ms-flex:100%;flex:100%;width:55vw;margin:auto}.home .sideBar .sideBarMenu a{font-size:4vw}.home .mainBody{-webkit-box-flex:100%;-ms-flex:100%;flex:100%}.sideBarTitle{height:7vw}.sideBarTitle p{font-size:4vw}.sideBar .sideBarMenu{height:70vw;width:100%}.adminMainBody .items{width:55vw;height:40vw}.adminMainBody .viewItem{width:55vw;height:40vw}.adminMainBody .viewItem .itemTitle{height:7vw}.adminMainBody .viewItem .itemTitle p{font-size:4vw}.adminMainBody .viewItem .itemContentLeft{height:16vw}.adminMainBody .viewItem .itemContentLeft p{font-size:2.5vw}.adminMainBody .viewItem .itemContentLeft h3{font-size:4vw}.adminMainBody .viewItem .itemContentRight{height:16vw}.adminMainBody .viewItem .itemContentRight p{font-size:2.5vw}.adminMainBody .viewItem .itemContentRight h3{font-size:4vw}.adminMainBody .viewItem .itemFooter{height:9vw}.adminMainBody .viewItem .itemFooter .itemFooterTitle{font-size:4.5vw}}
