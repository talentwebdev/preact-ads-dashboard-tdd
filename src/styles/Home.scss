%sam1 {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
}

%sam2 {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

%sam3 {
    text-align: left;
    margin-left: 5px;
    line-height: 5px;
}

%sam4 {
    height: 4.5vw;
    max-height: 20vw;
    margin-top: 1vw;
    margin-left: 5px;
    margin-right: 5px;
    width: 50%;
    padding: 5px 0;
    border-radius: 10px;
    box-shadow: 0px 3px 15px rgba(0, 0, 0, 0.5);
    p {
        @extend %sam3;
        font-size: 0.8vw;
    }
    h3 {
        @extend %sam3;
        font-size: 1.2vw;
    }
}

* {
    box-sizing: border-box;
}

.home {
    display: flex;
    width: 72%;
    margin: auto;
    margin-top: 2vw;
}

.sideBar {
    background-color: #fff;
    @extend %sam2;
    flex: 22%;
    border: 2px solid #0f4791;
    border-radius: 5px;
    max-height: 80vw;
    height: 25vw;
    margin-top: 4px;
}

.mainBody {
    display: flex;
    align-items: center;
    justify-content: center;
    flex: 78%;
    flex-wrap: wrap;
}

.sideBarTitle {
    margin-top: 0;
    background-color: #4d1936;
    height: 2.5vw;
    max-height: 10vw;
    width: 100%;
    overflow: hidden;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
    @extend %sam2;
    p {
        color: #fff;
        font-weight: 600;
        font-size: 1.2vw;
    }
}

.sideBarMenu {
    @extend %sam2;
    height: 100%;
    width: 100%;
    a {
        color: #4D1936;
        text-decoration: none;
        font-size: 1.3vw;
        font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
        padding: 0.2vw;
    }
    a:hover {
        background-color: gray;
        width: 100%;
    }
}

.adminMainBody {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    place-items: center;
    height: auto;
    align-items: center;
    place-items: center;
}

.adminMainBody .items {
    width: 18vw;
    height: 12.4vw;
    border: 2px solid #0f4791;
    border-radius: 10px;
    margin: 3px;
    box-shadow: 0px 3px 15px rgba(0, 0, 0, 0.5);
}

.adminMainBody {
    .viewItem {
        width: 18vw;
        height: 12.4vw;
        border: 2px solid #0f4791;
        border-radius: 10px;
        margin: 3px;
        box-shadow: 0px 3px 15px rgba(0, 0, 0, 0.5);
        overflow: hidden;
        .itemTitle {
            background-color: #4D1936;
            @extend %sam2;
            height: 2.4vw;
            p {
                color: #fff;
                font-size: 1.2vw;
                font-weight: 600;
            }
        }
    }
}

.itemContent {
    @extend %sam1;
    .itemContentLeft {
        @extend %sam4;
    }
    .itemContentRight {
        @extend %sam4;
    }
}

.itemFooter {
    @extend %sam1;
    height: 2.5vw;
    background-color: #E5D9B3;
    margin-top: 1vw;
    .itemFooterTitle {
        color: #4D1936;
        @extend %sam1;
        justify-content: flex-end;
        flex: 70%;
        font-size: 1.4vw;
        font-weight: 600;
        font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
        padding: 0.2vw;
    }
    .itemFooterIcon {
        @extend %sam1;
        flex: 30%;
        color: #4D1936;
    }
}

@media only screen and (max-width: 1450px) {
    .sideBar {
        height: 35vw;
    }
    .sideBarMenu {
        width: 19vw;
    }
    .sideBarTitle {
        height: 3vw;
    }
    .sideBarTitle p {
        font-size: 1.7vw;
    }
    .sideBar .sideBarMenu a {
        font-size: 1.5vw;
        padding: 0.5vw;
    }
    .adminMainBody {
        grid-template-columns: 1fr 1fr;
        .items {
            width: 26vw;
            height: 18vw;
        }
        .viewItem {
            width: 26vw;
            height: 18vw;
            .itemTitle {
                height: 3vw;
                p {
                    font-size: 1.7vw;
                }
            }
            .itemContentLeft {
                height: 7vw;
                p {
                    font-size: 1.2vw;
                }
                h3 {
                    font-size: 1.8vw;
                }
            }
            .itemContentRight {
                height: 7vw;
                p {
                    font-size: 1.2vw;
                }
                h3 {
                    font-size: 1.8vw;
                }
            }
            .itemFooter {
                height: 4vw;
                .itemFooterTitle {
                    font-size: 2vw;
                }
            }
        }
    }
}

@media only screen and (max-width: 1250px) {
    .sideBar .sideBarMenu a {
        font-size: 1.5vw;
    }
    .adminMainBody {
        grid-template-columns: 1fr 1fr;
        .items {
            width: 26vw;
            height: 15vw;
        }
        .viewItem {
            width: 26vw;
            height: 15vw;
            .itemContentLeft {
                height: 5vw;
                p {
                    font-size: 1.0vw;
                }
                h3 {
                    font-size: 1.5vw;
                }
            }
            .itemContentRight {
                height: 5vw;
                p {
                    font-size: 1.0vw;
                }
                h3 {
                    font-size: 1.5vw;
                }
            }
            .itemFooter {
                height: 3.5vw;
                .itemFooterTitle {
                    font-size: 2vw;
                }
            }
        }
    }
}

@media only screen and (max-width: 950px) {
    .sideBar {
        height: 50vw;
        margin-left: 2vw;
    }
    .sideBarTitle {
        height: 5vw;
        p {
            font-size: 2.5vw;
        }
    }
    .sideBarMenu {
        width: 27vw;
    }
    .sideBar .sideBarMenu a {
        font-size: 2.5vw;
    }
    .adminMainBody {
        grid-template-columns: 1fr;
        .items {
            width: 37vw;
            height: 27vw;
        }
        .viewItem {
            width: 37vw;
            height: 27vw;
            .itemTitle {
                height: 5vw;
                p {
                    font-size: 2.5vw;
                }
            }
            .itemContentLeft {
                height: 10vw;
                margin-top: 1.2vw 0vw;
                margin-bottom: 0.5vw;
                p {
                    font-size: 1.8vw;
                }
                h3 {
                    font-size: 2.5vw;
                }
            }
            .itemContentRight {
                height: 10vw;
                margin-top: 1.2vw;
                margin-bottom: 0.5vw;
                p {
                    font-size: 1.8vw;
                }
                h3 {
                    font-size: 2.5vw;
                }
            }
            .itemFooter {
                height: 6vw;
                .itemFooterTitle {
                    font-size: 3vw;
                }
            }
        }
    }
}

@media only screen and (max-width: 600px) {
    .home {
        display: flex;
        flex-direction: column;
        .sideBar {
            flex: 100%;
            width: 55vw;
            margin: auto;
            .sideBarMenu a {
                font-size: 4vw;
            }
        }
        .mainBody {
            flex: 100%;
        }
    }
    .sideBarTitle {
        height: 7vw;
        p {
            font-size: 4vw;
        }
    }
    .sideBar .sideBarMenu {
        height: 70vw;
        width: 100%;
    }
    .adminMainBody {
        .items {
            width: 55vw;
            height: 40vw;
        }
        .viewItem {
            width: 55vw;
            height: 40vw;
            .itemTitle {
                height: 7vw;
                p {
                    font-size: 4vw;
                }
            }
            .itemContentLeft {
                height: 16vw;
                p {
                    font-size: 2.5vw;
                }
                h3 {
                    font-size: 4vw;
                }
            }
            .itemContentRight {
                height: 16vw;
                p {
                    font-size: 2.5vw;
                }
                h3 {
                    font-size: 4vw;
                }
            }
            .itemFooter {
                height: 9vw;
                .itemFooterTitle {
                    font-size: 4.5vw;
                }
            }
        }
    }
}