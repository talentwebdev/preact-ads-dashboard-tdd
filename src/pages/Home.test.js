import { shallow } from "enzyme";
import Home from "./Home";

describe("Home", () => {
  let homeContainer;

  beforeEach(() => (homeContainer = shallow(<Home />)));

  it("should render the div element", () => {
    expect(homeContainer.find("div")).toBeTruthy();
  });
  it("renders Home correctly", () => {
    expect(homeContainer).toMatchSnapshot();
  });
});
