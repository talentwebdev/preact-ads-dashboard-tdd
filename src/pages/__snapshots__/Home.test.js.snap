// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders Home correctly 1`] = `ShallowWrapper {}`;
